lte/gateway/c/session_manager/LocalEnforcer.cpp

```diff
@@ -1150,31 +1164,31 @@ void LocalEnforcer::update_charging_credits(
     for (const auto& session : it->second) {
       std::string sid                = session->get_session_id();
       SessionStateUpdateCriteria& uc = session_update[imsi][sid];
-      bool is_redirected =
-          session->is_credit_state_redirected(CreditKey(credit_update_resp));
+      bool is_final_action_state =
+          session->is_credit_in_final_unit_state(CreditKey(credit_update_resp));
```

Comment:
You don't need that variable. You can use it on line 1173 directly since the name it is pretty self explanatory
```
if (session->is_credit_in_final_unit_state(CreditKey(credit_update_resp))) 
```

Reply:
The variable is_final_action_state (and now restrict_rules) is there to store the session state prior calling receive_charging_credit.
This latest will override those values.